SentinelOne/cloud_funnel2.0/ingest/parser.yml

```
@@ -0,0 +1,736 @@
+    external:
+      name: grok.match
+      properties:
+        input_field: '{{json_event.message.get("registry.keyPath")}}'
+        output_field: key
+        pattern: '%{GREEDYDATA}[\\\/]+%{GREEDYDATA:name}$'
+
+  - name: set_meta_fields
+
+  - name: set_common_fields
+
+  - name: set_process_fields
+    filter: '{{json_event.message.get("src.process.pid") != None}}'
+
+  - name: set_parent_fields
+    filter: '{{json_event.message.get("src.process.parent.pid") != None}}'
+
+  - name: set_target_process_fields
+    filter: '{{json_event.message.get("tgt.process.pid") != None}}'
+
+  - name: set_osSrc_fields
+    filter: '{{json_event.message.get("osSrc.process.pid") != None}}'
+
+  - name: set_osSrc_parent_fields
+    filter: '{{json_event.message.get("osSrc.process.parent.pid") != None}}'
+
+  - name: set_login_fields
+    filter: '{{json_event.message.get("event.category") == "logins"}}'
+
+  - name: set_ip_fields
+    filter: '{{json_event.message.get("src.ip.address") != None or json_event.message.get("dst.ip.address") != None}}'
+
+  - name: set_dns_fields
+    filter: '{{json_event.message.get("event.category") == "dns"}}'
+
+  - name: set_module_load_fields
+    filter: '{{json_event.message.get("event.category") == "module"}}'
+
+  - name: set_driver_load_fields
+    filter: '{{json_event.message.get("event.category") == "driver"}}'
+
+  - name: set_file_fields
+    filter: '{{json_event.message.get("tgt.file.path") != None}}'
+
+  - name: set_reg_value_fields
+    filter: '{{json_event.message.get("registry.value") != None}}'
+
+  - name: set_reg_key_fields
+    filter: '{{json_event.message.get("registry.keyPath") != None}}'
+
+  - name: set_script_fields
+    filter: '{{json_event.message.get("event.category") == "command_script"}}'
+
+  - name: set_http_fields
+    filter: '{{json_event.message.get("event.category") == "url"}}'
+
+  - name: set_indicator_fields
+    filter: '{{json_event.message.get("event.category") == "indicators"}}'
+
+stages:
+
+  set_meta_fields:
+    actions:
+      - set:
+          agent.version: '{{json_event.message.get("agent.version")}}'
+          deepvisibility.agent.managment_url: '{{json_event.message.get("mgmt.url")}}'
+          deepvisibility.agent.trace_id: '{{json_event.message.get("trace.id")}}'
+          deepvisibility.agent.uuid: '{{json_event.message.get("agent.uuid")}}'
+          deepvisibility.event.type: '{{json_event.message.get("event.type")}}'
+          deepvisibility.event.category: '{{json_event.message.get("event.category")}}'
+          event.action: '{{json_event.message.get("event.type")}}'
+          deepvisibility.host.os.revision: '{{json_event.message.get("mgmt.osRevision")}}'
+          host.name: '{{json_event.message.get("endpoint.name")}}'
+          host.os.family: '{{json_event.message.get("endpoint.os")}}'
+          host.os.name: '{{json_event.message.get("os.name")}}'
+          host.type: '{{json_event.message.get("endpoint.type")}}'
```

I suggest adding the following fields:
```suggestion
          host.type: '{{json_event.message.get("endpoint.type")}}'
          event.dataset: 'cloud-funnel-2.0'
          observer.vendor: 'SentinelOne'
```